src/cpu/aarch64/acl_gemm_convolution.hpp

```diff
@@ -105,13 +107,23 @@ struct acl_gemm_convolution_fwd_t : public primitive_t {
                 "gemm:acl", acl_gemm_convolution_fwd_t, USE_GLOBAL_SCRATCHPAD);
 
         status_t init(engine_t *engine) {
-            bool ok = true && is_fwd()
+            using namespace data_type;
+            using smask_t = primitive_attr_t::skip_mask_t;
+
+            bool ok = is_fwd()
                     && set_default_alg_kind(alg_kind::convolution_direct)
-                    && expect_data_types(data_type::f32, data_type::f32,
-                            data_type::f32, data_type::f32, data_type::f32)
+                    && expect_data_types(
+                            src_type, wei_type, bia_type, dst_type, undef)
                     && !has_zero_dim_memory()
-                    && attr()->has_default_values(
-                            primitive_attr_t::skip_mask_t::post_ops)
+                    && IMPLICATION(with_bias(),
+                            IMPLICATION(
+                                    src_type == u8, bias_md_.data_type == s32))
+                    && IMPLICATION(src_type == f32, bias_md_.data_type == f32)
```

Comment:
this will disable f32 convolution w/o bias:
```
src_type == f32 // true
bias_md_.data_type == undef // != f32 ==> false
```

Reply:
Oh yes, I understand my mistake there. Fixing now.